src/anomalib/metrics/per_image/binclf_curve.py

```diff
@@ -0,0 +1,178 @@
+"""Binary classification curve (torch interface).
```

Comment:
i tested all these variants back then
numba gives a good improvement, it is not neglectable
as for the tensor-ndarray-tensor conversion, it is also faster than using only torch

(from what i saw few months ago in the code and my own tests)
torchmetrics has two methods for computing this (both with cuda available)
A)  a fast one that computes the curves with all thresholds possible (ie all score values seen)
B) a slower one that takes pre-defined threholds

pimo needs B because all images are evaluated at the same thresholds
in that case, the methods here (even without numba) are faster because torch's and torchmetrics' stuff dont deal with computing same thresholds on instance-scoped binary classification matrices

i can find detailed results from my test again, but just for a reference (in the paper)

![image](https://github.com/openvinotoolkit/anomalib/assets/24547377/f704818d-7366-414b-9181-f25052f00d73)

that numba method (+ the tensor-ndarray conversion + just-in-time compilation) is beating torchmetrics AUROC on GPU

Reply:
i am resolving this one because it is a duplicate of https://github.com/openvinotoolkit/anomalib/pull/1726#discussion_r1654663073